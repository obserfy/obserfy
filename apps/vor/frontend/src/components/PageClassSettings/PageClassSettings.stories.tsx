import { FC } from "react"
import PageClassSettings from "./PageClassSettings"

export default {
  title: "Core/PageClassSettings",
  component: PageClassSettings,
  parameters: {
    componentSubtitle: "Just a simple PageClassSettings",
  },
}

export const Basic: FC = () => <PageClassSettings />
