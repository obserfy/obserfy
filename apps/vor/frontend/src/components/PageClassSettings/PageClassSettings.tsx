import { Trans } from "@lingui/macro"
import { StaticImage } from "gatsby-plugin-image"
import { FC } from "react"
import { Box, Button, Card, Flex } from "theme-ui"

import useGetSchoolClasses from "../../hooks/api/classes/useGetSchoolClasses"
import { ADMIN_URL, EDIT_CLASS_URL, NEW_CLASS_URL } from "../../routes"
import BackNavigation from "../BackNavigation/BackNavigation"
import { Link } from "../Link/Link"

import LoadingPlaceholder from "../LoadingPlaceholder/LoadingPlaceholder"
import Typography from "../Typography/Typography"

export const PageClassSettings: FC = () => {
  const classes = useGetSchoolClasses()

  const haveNoClass = classes.status === "success" && classes.data?.length === 0

  return (
    <Flex
      sx={{
        flexDirection: "column",
        maxWidth: "maxWidth.md",
      }}
      mx="auto"
    >
      <BackNavigation to={ADMIN_URL} text="Settings" />
      {classes.status === "loading" && <LoadingState />}
      {haveNoClass && <NoClassPlaceholder />}
      {(classes.data?.length ?? 0) > 0 && (
        <Flex sx={{ alignItems: "center" }} m={3}>
          <Typography.H4
            mr="auto"
            sx={{
              lineHeight: 1,
            }}
          >
            <Trans>Classes</Trans>
          </Typography.H4>
          <Link to={NEW_CLASS_URL}>
            <Button>
              <Trans>New</Trans>
            </Button>
          </Link>
        </Flex>
      )}
      {classes.data?.map(({ id, name }) => (
        <Link key={id} to={EDIT_CLASS_URL(id)}>
          <Card mx={3} mb={2} p={3}>
            <Typography.Body>{name}</Typography.Body>
          </Card>
        </Link>
      ))}
    </Flex>
  )
}

const NoClassPlaceholder: FC = () => (
  <Flex
    m={3}
    pt={4}
    sx={{
      flexDirection: "column",
      alignItems: "center",
    }}
  >
    <StaticImage
      src="../../images/calendar-colour.png"
      width={230}
      alt="a person pointing at a calendar"
      placeholder="blurred"
    />
    <Typography.Body my={4} mx={4} sx={{ textAlign: "center", maxWidth: 300 }}>
      <Trans>
        Tell us about your classes, We&apos;ll help you track your students
        attendance.
      </Trans>
    </Typography.Body>
    <Link to={NEW_CLASS_URL}>
      <Button>
        <Trans>New Class</Trans>
      </Button>
    </Link>
  </Flex>
)

const LoadingState: FC = () => (
  <Box m={3}>
    <LoadingPlaceholder sx={{ width: "20rem", height: 48 }} mb={3} />
    <LoadingPlaceholder sx={{ width: "100%", height: 62 }} mb={2} />
    <LoadingPlaceholder sx={{ width: "100%", height: 62 }} mb={2} />
    <LoadingPlaceholder sx={{ width: "100%", height: 62 }} mb={2} />
  </Box>
)

export default PageClassSettings
