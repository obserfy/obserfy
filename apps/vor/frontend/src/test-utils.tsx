/* eslint-disable import/no-extraneous-dependencies */
import { FC, ReactElement } from "react"
import { QueryClient, QueryClientProvider } from "react-query"
import { ThemeProvider, Box, Theme } from "theme-ui"
import { render, RenderOptions, RenderResult } from "@testing-library/react"
import {
  createHistory,
  createMemorySource,
  LocationProvider,
} from "@reach/router"
import { I18nProvider } from "@lingui/react"
import { i18n } from "@lingui/core"
import { en } from "make-plural/plurals"
import theme from "./gatsby-plugin-theme-ui"
// @ts-ignore
import enCatalog from "../i18n/lingui/en/messages.js"

// any used here because the catalogs are generated b lingui
// not our responsibility
i18n.loadLocaleData("en", { plurals: en })
i18n.load("en", enCatalog.messages as any)
i18n.activate("en")

const queryClient = new QueryClient()
const history = createHistory(createMemorySource("/"))

const AllProviders: FC = ({ children }) => {
  return (
    <I18nProvider i18n={i18n}>
      <QueryClientProvider client={queryClient}>
        <LocationProvider history={history}>
          <ThemeProvider theme={theme as Theme}>
            <Box sx={{ fontSize: [16, 20] }}>{children}</Box>
          </ThemeProvider>
        </LocationProvider>
      </QueryClientProvider>
    </I18nProvider>
  )
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "queries">
): RenderResult => render(ui, { wrapper: AllProviders, ...options })

// re-export everything
export * from "@testing-library/react"

// override render method
export { customRender as render }
