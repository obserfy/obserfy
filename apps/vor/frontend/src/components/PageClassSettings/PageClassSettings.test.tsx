import { render } from "../../test-utils"
import { Basic } from "./PageClassSettings.stories"

describe("PageClassSettings", () => {
  it("should render correctly", () => {
    const { container } = render(<Basic />)
    expect(container).toMatchSnapshot()
  })
})
