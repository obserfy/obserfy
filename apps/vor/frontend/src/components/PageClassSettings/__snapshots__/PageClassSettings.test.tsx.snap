// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageClassSettings should render correctly 1`] = `
@keyframes animation-0 {
  0% {
    opacity: 0.05;
  }

  50% {
    opacity: 0.2;
  }

  100% {
    opacity: 0.05;
  }
}

@keyframes animation-0 {
  0% {
    opacity: 0.05;
  }

  50% {
    opacity: 0.2;
  }

  100% {
    opacity: 0.05;
  }
}

.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  font-size: 16px;
}

@media screen and (min-width: 40em) {
  .emotion-0 {
    font-size: 20px;
  }
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-left: auto;
  margin-right: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  max-width: 700px;
}

.emotion-2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-left: -8px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-3 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  max-width: 100%;
  height: auto;
  margin: 16px;
  margin-right: 4px;
  fill: var(--theme-ui-colors-icon);
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 24px;
  height: 24px;
  fill: var(--theme-ui-colors-textMediumEmphasis);
}

.emotion-4 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  color: var(--theme-ui-colors-text);
  font-size: 14px;
  text-rendering: optimizeLegibility;
  line-height: 1.5;
  margin-bottom: 0;
  color: var(--theme-ui-colors-textMediumEmphasis);
}

.emotion-5 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin: 16px;
}

.emotion-6 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-bottom: 16px;
  border-radius: 6px;
  -webkit-animation: 4s infinite animation-0;
  animation: 4s infinite animation-0;
  background-color: #a5a5a5;
  width: 20rem;
  height: 48px;
}

.emotion-7 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-bottom: 8px;
  border-radius: 6px;
  -webkit-animation: 4s infinite animation-0;
  animation: 4s infinite animation-0;
  background-color: #a5a5a5;
  width: 100%;
  height: 62px;
}

<div>
  <div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      <a
        href="/dashboard/admin"
        style="display: block;"
      >
        <div
          class="emotion-2"
        >
          <div
            class="emotion-3"
          />
          <p
            class="emotion-4"
          >
            Settings
          </p>
        </div>
      </a>
      <div
        class="emotion-5"
      >
        <div
          class="emotion-6"
        />
        <div
          class="emotion-7"
        />
        <div
          class="emotion-7"
        />
        <div
          class="emotion-7"
        />
      </div>
    </div>
  </div>
</div>
`;
